### PR TITLE
feat: Add new admin command to print Argo CD initial password (11117)

### LINKED ISSUE #11117: Add admin command to get initial password
# Summary

Add an `argocd admin initial-password` command to get the initial password.

# Motivation

It's annoying to remember/use the `kubectl` command from the docs. That command also doesn't explicitly remind you to change the initial password.

# Proposal

The command should use your current kube config to pull down the initial password. Then it should strongly encourage you to change that password.

### PR DESCRIPTION
**Pull request by iam-veeramalla**
Signed-off-by: iam-veeramalla <abhishek.veeramalla@gmail.com>

Note on DCO:

If the DCO action in the integration test fails, one or more of your commits are not signed off. Please click on the *Details* link next to the DCO action for instructions on how to resolve this.

Checklist:

* [x] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo-cd/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this does not need to be in the release notes.
* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [x] I've included "Closes [ISSUE #]" or "Fixes [ISSUE #]" in the description to automatically close the associated issue.
* [x] I've updated both the CLI and UI to expose my feature, or I plan to submit a second PR with them.
* [x] Does this PR require documentation updates?
* [x] I've updated documentation as required by this PR.
* [ ] Optional. My organization is added to USERS.md.
* [x] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/tree/master/community#contributing-to-argo)
* [ ] I have written unit and/or e2e tests for my change. PRs without these are unlikely to be merged.
* [x] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/latest/developer-guide/ci/)). 

Fixes #11117 
Closes #11117

### CODE REVIEW

File: cmd/argocd/commands/admin/initial_password.go [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+			namespace, _, err := clientConfig.Namespace()
+			errors.CheckError(err)
+
+			kubeClientset := kubernetes.NewForConfigOrDie(config)
+			secret, err := kubeClientset.CoreV1().Secrets(namespace).Get(context.Background(), initialPasswordSecretName, v1.GetOptions{})
+			errors.CheckError(err)
+
+			if initialPass, ok := secret.Data["password"]; ok {
+				fmt.Println(string(initialPass))
+				fmt.Println("\n This password must be only used for first time login. We strongly recommend you to update the password using `argocd account update-password`.")
```
Comment (crenshaw-dev) [inline cmd/argocd/commands/admin/initial_password.go:39]:
```suggestion
				fmt.Println("\n This password must be only used for first time login. We strongly recommend you update the password using `argocd account update-password`.")
```

**Review by crenshaw-dev — CHANGES_REQUESTED** [review]
Thanks @iam-veeramalla! One small grammar tweak.

Could you also update the quick start guide to reference this new command instead of the `kubectl` command?

**Comment by crenshaw-dev** [comment]
Can you run CLI codegen again?

**Comment by iam-veeramalla** [comment]
My mac is giving me hard time in running the codegen and other make targets :( . Hope this commit will fix the failing CI.

**Comment by iam-veeramalla** [comment]
@crenshaw-dev got the code-gen working now. PTAL

**Comment by iam-veeramalla** [comment]
@crenshaw-dev just in case you missed my previous comment. 

Fixed the failing CI and rebased the code.

File: docs/getting_started.md [hunk 2]
```diff
@@ -78,10 +78,10 @@ The API server can then be accessed using https://localhost:8080
 The initial password for the `admin` account is auto-generated and stored as
 clear text in the field `password` in a secret named `argocd-initial-admin-secret`
 in your Argo CD installation namespace. You can simply retrieve this password
-using `kubectl`:
+using `argocd admin command`:
```
Comment (crenshaw-dev) [inline docs/getting_started.md:81]:
```suggestion
using the `argocd` CLI:
```

File: cmd/argocd/commands/admin/initial_password.go [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+const initialPasswordSecretName = "argocd-initial-admin-secret"
+
+// NewInitialPasswordCommand defines a new command to retrieve Argo CD initial password.
+func NewInitialPasswordCommand() *cobra.Command {
+	var (
+		clientConfig clientcmd.ClientConfig
+	)
+	var command = cobra.Command{
+		Use:   "initial-password",
+		Short: "Prints initial password to login into Argo CD for the first time",
```
Comment (crenshaw-dev) [inline cmd/argocd/commands/admin/initial_password.go:25]:
```suggestion
		Short: "Prints initial password to log in to Argo CD for the first time",
```

File: docs/user-guide/commands/argocd_admin_initial-password.md [hunk 4]
```diff
@@ -0,0 +1,59 @@
+## argocd admin initial-password
+
+Prints initial password to login into Argo CD for the first time
```
Comment (crenshaw-dev) [inline docs/user-guide/commands/argocd_admin_initial-password.md:3]:
```suggestion
Prints initial password to log in to Argo CD for the first time
```

**Review by crenshaw-dev — CHANGES_REQUESTED** [review]

**Review by crenshaw-dev — APPROVED** [review]

File: docs/user-guide/commands/argocd_admin.md [hunk 5]
```diff
@@ -43,6 +43,7 @@ argocd admin [flags]
 * [argocd admin dashboard](argocd_admin_dashboard.md)	 - Starts Argo CD Web UI locally
 * [argocd admin export](argocd_admin_export.md)	 - Export all Argo CD data to stdout (default) or a file
 * [argocd admin import](argocd_admin_import.md)	 - Import Argo CD data from stdin (specify `-') or a file
+* [argocd admin initial-password](argocd_admin_initial-password.md)	 - Prints initial password to login into Argo CD for the first time
```
Comment (crenshaw-dev) [inline docs/user-guide/commands/argocd_admin.md:46]:
```suggestion
* [argocd admin initial-password](argocd_admin_initial-password.md)	 - Prints initial password to log in to Argo CD for the first time
```

**Comment by ofirshtrull** [comment]
I don't know if I'm missing something but, how is this helpful? if you ran `argocd login` that means that you already have the password in hand...
so how does this benefit the users?

**Comment by gopikrishna7** [comment]
@ofirshtrull , Before `argocd login`  you can run this command to get the initial admin password. Using this initial admin password you can login to ArgoCD UI or CLI.

**Pull request merged**